nucypher/policy/conditions/offchain.py

```diff
@@ -0,0 +1,168 @@
+from typing import Any, Optional, Tuple
+
+import requests
+from jsonpath_ng.exceptions import JsonPathLexerError, JsonPathParserError
+from jsonpath_ng.ext import parse
+from marshmallow import fields, post_load, validate
+from marshmallow.fields import Field
+
+from nucypher.policy.conditions.base import AccessControlCondition
+from nucypher.policy.conditions.exceptions import (
+    ConditionEvaluationFailed,
+    InvalidCondition,
+    InvalidConditionLingo,
+)
+from nucypher.policy.conditions.lingo import ConditionType, ReturnValueTest
+from nucypher.policy.conditions.utils import CamelCaseSchema
+from nucypher.utilities.logging import Logger
+
+
+class JSONPathField(Field):
+    default_error_messages = {
+        "invalidType": "Expression of type '{type(value)}' is not valid for JSONPath",
+        "invalid": "'{value}' is not a valid JSONPath expression",
+    }
+
+    def _deserialize(self, value, attr, data, **kwargs):
+        if not isinstance(value, str):
+            raise self.make_error("invalidType", input=value)
+        try:
+            parse(value)
+        except (JsonPathLexerError, JsonPathParserError):
+            raise self.make_error("invalid", value=value)
+        return value
+
+
+class HTTPSField(Field):
+    default_error_messages = {
+        "invalid": "'{value}' is not a valid HTTPS endpoint",
+    }
+
+    def _deserialize(self, value, attr, data, **kwargs):
+        if not value.startswith("https://"):
```

Comment:
Do we need a more complete validation here? E.g. `https://https://https://foofoofoo` satisfies current validation. It seems there's packages for this (e.g. https://yozachar.github.io/pyvalidators/stable/api/url/)

Reply:
If more robust validation is needed, I'd prefer to make a simple regex here vs. introducing a new dependency.  Alternatively I might be able to use some standlib url parsing tools.  In any case, I don't think it's a serious concern since only valid URLs will be usable by requests.

Reply:
Oh that's a nice find!